### PR TITLE
Filtering, metadata docs

### PR DESCRIPTION
**Pull request by dankraw**
...smuggling a warning suppression. :>

### CODE REVIEW

**Comment by wojtkiewicz** [comment]
👍

File: docs/docs/user/subscribing.md [hunk 1]
```diff
@@ -152,6 +173,71 @@ is growing. First things first, you should check subscription metrics for signs
...
+## Additional headers
+
+Each subscription can define a number of additional `headers` that will be added to every HTTP request when sending messages.
+They can be useful on test environments to pass security tokens or on production to communicate with some legacy systems that 
+require custom headers.
+
+## Endpoint address resolver metadata
+
+Custom implementation of Consumer's `EndpointAddressResolver` interface can make use of provided `endpointAddressResolverMetadata` 
+for selecting address resolving strategies. This object deserializes to `Map<String, Object>` and may contain any data needed,
```
Comment (druminski) [inline docs/docs/user/subscribing.md]: "This object deserializes to". What is deserialized? If endpointAddressResolverMetadata then maybe "This object is deserialized to ..." will be more clear.  Beside that, LGTM

**Comment by druminski** [comment]
👍

File: docs/docs/user/subscribing.md (hunk 1 above)
Comment (wojtkiewicz) [inline docs/docs/user/subscribing.md]: done

**Comment by adamdubiel** [comment]
👍

**Pull request merged**